src/middleware.coffee

```diff
@@ -0,0 +1,27 @@
+basic_auth = require('basic-auth')
+
+
+module.exports = 
+
+  auth: (conf) ->
+    SYSTEM_USER = conf.COUCHDB.SYSTEM_USER
+    (req, resp, next) ->
+      # look for admin credentials in basic auth, and if valid, login user as admin.
+      credentials = basic_auth(req);
+      if conf.DEV = 'true' or 
```

Comment:
Is this supposed to allow regular user credentials or just SYSTEM_USER or nothing? Looking at the syntax it seems to kick you out if you have valid user credentials that are not admin and are not in conf.DEV.